backend/api/main.py

```diff
@@ -182,13 +198,33 @@ async def add_datapoint(
         UniqueViolationError: If the object_id of the datapoint already exists in the database, a 409 error will be raised.
         Exception: If some other error occurs, a 500 error will be raised.
     """
-    datapoint.object_id = str(uuid4())
-    if datapoint.matchDatapoint and (
-            datapoint.entity_id is None or datapoint.attribute_name is None
+    # Generate a new 6-character object_id if not provided
+    if datapoint.object_id is None:
+        while True:
+            new_id = str(uuid4())[:6]
+            existing = await conn.fetchrow(
+                """SELECT object_id FROM datapoints WHERE object_id=$1""",
+                new_id
+            )
+            if not existing:
+                datapoint.object_id = new_id
+                break
+
+    else:
+        # Check if the provided object_id is unique
+        existing = await conn.fetchrow(
+            """SELECT object_id FROM datapoints WHERE object_id=$1""",
+            datapoint.object_id
+        )
+        if existing:
+            raise HTTPException(status_code=409, detail="object_id already exists!")
+
+    if datapoint.connected and (
```

Comment:
Imop, users cannot manually change or specify `datapoint.connected`. This value should only be updated by the checking function.

Reply:
This is also fixed